### PR TITLE
244 gis

### PR DESCRIPTION
**Pull request by RobertClay**
Pull request for some features added for GIScience paper 

- simplifying lineplot code. making Rmd notebooks where you can render all lineplots quickly. Tried bash scripts as well but think notebook approach is much easier. 
- EPGC and EBSS energy policy interventions. just modifications of energy downlift. 
- Glasgow synthetic input population. Can have some percentage scale of glasgow that merges on WS3 data to have a synthetic component. Had to increase arc4 memory allocation to 5GB due to large file size. Some work needed to trim unnecessary columns in US_upscaling I think. 
- update maps to be able to use synthetic data spatial component directly. much faster map plotting. also multithreading for maps which is a lot quicker.
- tried to make a notebook for maps but this wont work until arm64 r-sf is on conda forge (at least for mac users). 
- arm64 config for mac users. The current environment really did not like me having a new version of R. 
- minor tweaks to maps for GIScience. Colour scheme and labels. 
- organise persistent data into subfolders for spatial data, rate table data. 

TODO BEFORE MERGING
- test lineplots functions work on other OS and with other data. Plots look wierd at the minute on dev but not sure if thats due to some misbehaviing transition model in dev or what. Need to be very sure the new code does what its supposed to. 
- tidying up some scrap ideas. Probabily should just bin the bash scripts they're not worth it. Need some dicussion on what to do what outcomes makefile as well. Its very messy atm.

### CODE REVIEW

**Comment by RobertClay** [comment]
closes #244

**Comment by RobertClay** [comment]
notebook for lineplots now uses relative imports. scrap ideas using bash scripts mostly cleaned up. still need to convert maps to notebook but low priority.

File: minos/validation/longitudnial_visualisation.R
Comment (ld-archer) [inline minos/validation/longitudnial_visualisation.R:1]: Duplicate of longitudinal_visualisation.R

File: plots/epcg_ebss_baseline_SF_12_aggs_by_year.pdf
Comment (ld-archer) [inline plots/epcg_ebss_baseline_SF_12_aggs_by_year.pdf:1]: This looks like a file generated by the map plotting scripts. Can we ignore this as it doesn't need to be tracked. Goes for all other new lineplot and map files.

File: minos/outcomes/sf12_difference_map.R
Comment (ld-archer) [inline minos/outcomes/sf12_difference_map.R:1]: Getting this error running `make epcg_ebss_map` with a freshly created conda environment:
```
Error in library(geojsonio) : there is no package called ‘geojsonio’
```

Does this package need to be added to environment.yml? I remember issues with some of these mapping packages in the past, does this need installing manually? Do we need to handle it separately as with Mac problems with pscl, ordinal above?

Comment (ld-archer) [reply minos/outcomes/sf12_difference_map.R:1]: Trying to install this package manually is causing all kinds of problems on my local machine also, missing apt packages and all sorts so not an easy thing to install manually.

**Review by ld-archer — CHANGES_REQUESTED** [review]
Left a few comments on some scripts with errors / suggested changes. Some of them would be fixed by merging development into the branch, others a bit of moving files around and renaming etc. No major issues although I haven't been able to test the epcg_ebss_map target due to the missing geojsonio package (and problems installing manually).

File: minos/outcomes/sf12_difference_map.R
Comment (RobertClay) [reply minos/outcomes/sf12_difference_map.R:1]: thought it was a good alternative but its not. removing it and sticking to r-sf package instead.

File: minos/data_generation/US_upscaling.py
[the remark this answers is not in the document]
Comment (RobertClay) [reply minos/data_generation/US_upscaling.py:1]: adding error message and files to git ignore.

**Comment by RobertClay** [comment]
@ld-archer all requested changes have been made.

**Comment by RobertClay** [comment]
@ld-archer This branch is now ready to merge into dev snapshot.

**Pull request closed without merging**